Frameworks/include/ABMultiValueInternal.h

```diff
@@ -0,0 +1,44 @@
+//******************************************************************************
+//
+// Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#pragma once
+
+#import <AddressBook/ABMultiValue.h>
+#import <Foundation/Foundation.h>
+
+@interface _ABMultiValue : NSObject
+
+- (id)initWithPropertyType:(ABPropertyType)propertyType;
+
+// Once the MultiValue is made immutable,
+// appendPairWithLabel:andValue: no longer does anything;
+// it returns YES if the appending occurred, or NO if the
+// MultiValue is immutable and no change was made.
```

Comment:
since this is logically a one way valve, why not just make it part of the init call? init could take an array/dictionary of pairs and then it would never need to deal with mutability.